app/server/lib/GranularAccess.ts

```diff
@@ -3174,7 +3174,13 @@ export class CensorshipInfo {
 
   public apply(a: DataAction) {
     const tableId = getTableId(a);
-    if (!STRUCTURAL_TABLES.has(tableId)) { return true; }
+    if (!STRUCTURAL_TABLES.has(tableId)
+      // Don't censor _grist_Pages, as that requires a finer approach.
```

Comment:
Mmmm showing or not showing pages isn't currently treated as a security issue, it is purely a rendering choice by the web client. Don't want to blur the language around security here. Nothing about the _grist_Pages *table* is censored, all clients have access to all information in this table. That's all we're concerned about here. Please just state that nothing in the _grist_Pages table is censored.

Reply:
We are defining what information is shared with everyone, by every channel (web socket or api) and what information is hidden from some. Currently, everything in the _grist_Pages table is shared with everyone. There are many tables. If you want to get into the business of explaining what is censored and why, The docstring for `CensorshipInfo` would probably be a better place.

How about apply()ing unconditionally, moving the `if (!STRUCTURAL_TABLES.has(tableId)) { return true; }` part into filter() after the ACL stuff, and changing the ACL stuff to be clearer that it applies specifically to `_grist_ACLRules` and `_grist_ACLResources` and possibly `_grist_Shares`? The implicit assumption in that code about what tables it applies to feels like the weak link here.